tensorflow/compiler/jit/xla_gpu_device.cc

```diff
@@ -33,8 +33,34 @@ class XlaGpuDeviceFactory : public DeviceFactory {
  public:
   Status CreateDevices(const SessionOptions& options, const string& name_prefix,
                        std::vector<std::unique_ptr<Device>>* devices) override;
+  // Returns a set containing the device ids contained in visible_device_list or
+  // -1 if the string is empty.
```

Comment:
Should we be returning a `StatusOr<optional<set<int>>>` instead of returning {-1}?

But also I'm confused, it looks like we never call this function when the input string is empty?  So it never actually returns nullopt or {-1}.  Seems like we could simplify its contract?

Reply:
Changed it to return optional.